### PR TITLE
Handle WebAuth on Extreme Controller without web form

### PR DESCRIPTION
**Pull request by nqb**
# Description
Handle WebAuth on Extreme Controller without web form

Web form usage still work but some browsers complain when they have to send a web form to 1.1.1.1/reg.php using HTTP (in place of HTTPS). Switching to HTTPS break everything.

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Improve WebAuth support on Extreme controllers

### CODE REVIEW

**Comment by nqb** [comment]
Not sure if I need to add a dedicated URL for WebAuth in constants.pm

**Comment by julsemaan** [comment]
> Not sure if I need to add a dedicated URL for WebAuth in constants.pm

I believe you do since your configuration in the Extreme controller will be to point the webauth URL to `/Extreme::Extreme_Controller` so that URL must be allow-listed by being in pf::web::constants

**Comment by nqb** [comment]
I made some adjustments and tests: it works as expected.

I need some confirmation from Extreme then we can merge (for v12 ?) if you agree.

I can also add some documentation.

**Comment by julsemaan** [comment]
I do think we need some doc on configuring this or at the very least a link to some existing external doc we can follow that gives the right result.

I do think it's mergeable for v12 given it has a very little footprint and is self-contained

**Comment by julsemaan** [comment]
@nqb, do you think you will be able to add some doc to this? Or link to a documentation published by Extreme we could follow for the controller/AP config?

**Comment by nqb** [comment]
I will do my best to provide this and get this PR merged for 12.1.0

**Comment by nqb** [comment]
Doc added, I checked AsciiDoctor display and it looks good. 

I also change name of module to be close to what we have for Aerohive.

**Comment by julsemaan** [comment]
@julsemaan, do a final review and merge

File: docs/network/networkdevice/extreme_networks_wireless.asciidoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    for authors, copyright and license information.
+
+////
+
+
+//=== Extreme Networks
+
+==== Access points AP305C (managed by Extreme Cloud IQ Controller)
+
+In such deployment, PacketFence communicate directly with access
```
Comment (julsemaan) [inline docs/network/networkdevice/extreme_networks_wireless.asciidoc]:
```suggestion
In such deployment, PacketFence communicates directly with the access
```

File: docs/network/networkdevice/extreme_networks_wireless.asciidoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+points using RADIUS. Extreme Cloud IQ controller is only used to configure
```
Comment (julsemaan) [inline docs/network/networkdevice/extreme_networks_wireless.asciidoc]:
```suggestion
points using RADIUS. The Extreme Cloud IQ controller is only used to configure
```

File: docs/network/networkdevice/extreme_networks_wireless.asciidoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+access points in a central place.
+
+===== Web authentication
+
+====== Extreme Cloud IQ Controller
+
+On Extreme Cloud IQ Controller, you should have builtin IP Firewall policies:
```
Comment (julsemaan) [inline docs/network/networkdevice/extreme_networks_wireless.asciidoc]:
```suggestion
On the Extreme Cloud IQ Controller, there should already be two built-in IP Firewall policies:
```

File: docs/network/networkdevice/extreme_networks_wireless.asciidoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+* Redirect-only policy: this policy must have "redirect" rules
+* Internet-access-only policy
+
+On Extreme Cloud IQ Controller, create two user profiles:
```
Comment (julsemaan) [inline docs/network/networkdevice/extreme_networks_wireless.asciidoc]:
```suggestion
On the Extreme Cloud IQ Controller, create two user profiles:
```

File: docs/network/networkdevice/extreme_networks_wireless.asciidoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+** VLAN: 5
+** Firewall rules: Enabled
+** IP Firewall Name: Redirect-only policy
+** Redirect URL: http://192.168.1.5/Extreme::AP
+* Guest
+** VLAN: 5
+** Firewall rules: Enabled
+** IP Firewall Name: Internet-access-only policy
+
+On Extreme Cloud IQ Controller, create a wireless network with following configuration:
```
Comment (julsemaan) [inline docs/network/networkdevice/extreme_networks_wireless.asciidoc]:
```suggestion
Still, on the Extreme Cloud IQ Controller, create a wireless network with the following configuration:
```

**Comment by julsemaan** [comment]
Minor verbiage stuff in the docs, since @nqb is off, I'll apply them and merge

**Pull request merged**